src/ripple/app/misc/Transaction.h

```diff
@@ -66,79 +73,63 @@ class Transaction : public std::enable_shared_from_this<Transaction>,
     using ref = const pointer&;
 
     Transaction(
-        std::shared_ptr<STTx const> const&,
-        std::string&,
-        Application&) noexcept;
-
-    // The two boost::optional parameters are because SOCI requires
-    // boost::optional (not std::optional) parameters.
-    static Transaction::pointer
-    transactionFromSQL(
-        boost::optional<std::uint64_t> const& ledgerSeq,
-        boost::optional<std::string> const& status,
-        Blob const& rawTxn,
-        Application& app);
-
-    // The boost::optional parameter is because SOCI requires
-    // boost::optional (not std::optional) parameters.
-    static TransStatus
-    sqlTransactionStatus(boost::optional<std::string> const& status);
+        std::shared_ptr<STTx const> tx,
+        TransStatus status = NEW,
+        std::uint32_t ledgerSeq = 0) noexcept;
 
     std::shared_ptr<STTx const> const&
-    getSTransaction()
+    getSerializedTx() const noexcept
```

Comment:
Since this returns an `STTx` ptr, why not call the function `getSTTx()` or just `getTx()`?